DSignJar multiple file selection (issue #239)

Colbix opened the pull request:
The multiple file selection request, #239, is proceeding well. I would appreciate any feedback for the PR. One final dialog for error handling needs to be created for this to be complete and I would also like input on how best to implement.

My thinking is that ideally the errors should go to an error log but I don't know if one exists for KSE. The initial thought is to create an error dialog view with either a table or text area to dump the information. Thank you for your attention.

![image](https://user-images.githubusercontent.com/25093217/127246171-1d28b0de-3f16-46c5-800b-6f673f70ef3c.png)
![image](https://user-images.githubusercontent.com/25093217/127246282-9d258911-66ff-44a1-b0f0-147df826bbed.png)
![image](https://user-images.githubusercontent.com/25093217/127246500-1859421e-9b01-4528-9c20-e940c048d14f.png)
![image](https://user-images.githubusercontent.com/25093217/127247712-7f04b98e-210a-4d76-bb70-e0e90d761aa6.png)
![image](https://user-images.githubusercontent.com/25093217/127247277-1b6f575e-a96e-4676-9199-8f3136da1815.png)

kaikramer commented:
Alright, nice! I appreciate your diligence.

My feedback for the user interface:

- I think it is confusing that there are two ways to select the jars: Complete folder and (multiple) file selection. As a longtime KSE user I just clicked on the first browse button without reading the label text and was completely confused because I could not select or even see any files. I would simply remove the folder selection as it is superfluous anyway. If I want to select all jar files in a folder, I can simply do Ctrl-A or if there are other file types in that folder, sort by type and select them:
![grafik](https://user-images.githubusercontent.com/7840192/127773956-0b5e92b9-39d8-40b9-b8dc-9303f781e13c.png)
- If I select 4 jar files, then it would be nice to see something like "4 files selected" next to the browse button. Or even better: Depending on the number of selected files either show the file names if they fit in the text field or just the number if there are too many files.
![grafik](https://user-images.githubusercontent.com/7840192/127774065-b76e8538-1e88-4c59-ad56-ed67f2f31d5d.png)
![grafik](https://user-images.githubusercontent.com/7840192/127774166-5b2f20cd-adff-4bc5-bd53-c5e935646963.png)
- After the files are signed, again show either the number of successfully signed files (and maybe also the number of failures) or if there are not too many show a list of the signed files. Maybe like your suggestion to show a table with all the files and then a status (success/error info) next to them.
- Error handling: There is no error log in KSE and I think this would be weird and cumbersome for the user in a GUI application. As mentioned above I think the best option is to show a result dialog with an overview of the jar files and the result of the signature action.
- I think UI-wise everything else is fine (haven't looked in the code so far).

Colbix commented:
I removed the folder only selection although if there is a demand for the feature I think I have a better implementation so as not to confound the user.

I added a new error class for displaying multiple errors utilizing a jlist with a hashmap. The error handling should show how many files were successful and how many errors were found. The view button will show the errors from a selectable list.

![image](https://user-images.githubusercontent.com/25093217/128614938-d11042f1-1237-4e1b-a303-1b1091852c72.png)
![image](https://user-images.githubusercontent.com/25093217/128614944-76a75817-4a04-4088-a7b7-45bdb44a3805.png)
![image](https://user-images.githubusercontent.com/25093217/128614960-98a67c85-2f6d-4e3b-acd7-ed4cb97a4dfe.png)

Colbix commented:
Added a file count label to the DSignJar class.

![image](https://user-images.githubusercontent.com/25093217/128616419-4b6381f2-2660-4848-b8d8-6f85cd89cd36.png)

Colbix commented:
I examined the failed gradle checks, I believe the test was against the arguments in the SignJarAction class for the DSignJar class. The arguments for the class were changed however as I was committing individual classes the checks didn't pick up on the subsequent commit's.

kaikramer commented:
I think it's fine, but please change the texts for this dialog:

![grafik](https://user-images.githubusercontent.com/7840192/128645609-07008871-05ff-4159-8b12-581ed7c77b48.png)

"Error Collection" -> "Errors"
"Keys" -> "File"
"Key Value" -> "Error"

BTW, what is your DPI setting? And do you have other GUI issues like the cut-off icon in this dialog?:

![grafik](https://user-images.githubusercontent.com/25093217/128614960-98a67c85-2f6d-4e3b-acd7-ed4cb97a4dfe.png)

Colbix commented:
I updated the properties. My resolution is 1920x1080 @ 146.86 PPI. I receive the icon cut off for all the dialogs not just that one.

The-Lum commented on kse/src/org/kse/gui/actions/SignJarAction.java:
```diff
@@ -85,43 +85,88 @@ protected void doAction() {
+				String strView = res.getString("SignJarAction.ButtonView.message");
+				// set view and ok buttons
+				Object[] buttons = { strView, strOK };
+				// set file count
+				String fileCount = Integer.toString(inputJarFile.length);
+				// set exception count
+				String errorCount = Integer.toString(dSignJarSigning.getFileExceptions().size());
```
Hello @kaikramer and @Colbix,

- Is it possible to change the type of `fileCount` and `errorCount` from `String` to `Integer`?
_(especially for improve i18n on resources files)_

Could you make a PR?
or would you like that I make one?

The-Lum commented on kse/src/org/kse/gui/actions/SignJarAction.java:
```diff
@@ -85,43 +85,88 @@ protected void doAction() {
 			} else {
-				JarSigner.sign(inputJarFile, outputJarFile, privateKey, certs, signatureType, signatureName, signer,
-						digestType, tsaUrl, provider);
+				// set file count
+				String fileCount = Integer.toString(inputJarFile.length);
```
Hello @kaikramer and @Colbix,

_The same..._
- Is it possible to change the type of `fileCount` from `String` to `Integer`?
_(especially for improve i18n on resources files)_

Could you make a PR?
or would you like that I make one?

kaikramer commented:
@The-Lum I would like to simplify the code in this class a bit. So I will change the type of fileCount and errorCount as well.

kaikramer commented:
@The-Lum Done
